pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -146,26 +160,40 @@ func resolvePipeline(ctx context.Context, resolver remote.Resolver, name string,
 // An error is returned if the given object is not a
 // PipelineObject, trusted resources verification fails or if there is an error validating or upgrading an
 // older PipelineObject into its v1beta1 equivalent.
+// This method sets defaults and applies validation based on the API version of the referenced Pipeline.
 // TODO(#5541): convert v1beta1 obj to v1 once we use v1 as the stored version
 func readRuntimeObjectAsPipeline(ctx context.Context, obj runtime.Object, k8s kubernetes.Interface, refSource *v1beta1.RefSource, verificationPolicies []*v1alpha1.VerificationPolicy) (*v1beta1.Pipeline, error) {
+	// Inject a sentinel value into the context to skip validation of parameters and workspaces that may not yet be present
+	ctx = config.SkipValidationDueToPropagatedParametersAndWorkspaces(ctx, true)
```

Comment:
Instead of setting this, would it make more sense to reverse this condition and only set it when strict validation is needed? (it seems like this happens less often).

Alternatively, can we drop this altogether and only use the looser validation rules?

Reply:
I'm working on removing this flag altogether: https://github.com/tektoncd/pipeline/issues/6647